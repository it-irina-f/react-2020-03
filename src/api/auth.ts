export const sleep = (x: number) => new Promise((r) => setTimeout(r, x));

export const login = async (name: string) => {
  await sleep(1000);

  await localStorage.setItem("login", name);
};

export const logout = async () => {
  await sleep(1000);

  await localStorage.removeItem("login");
};

export const getUserSession = async () => {
  await sleep(2000);
  const login = await localStorage.getItem("login");
  return login;
};

export const isLoggedIn = async () => {
  const login = await getUserSession();
  return Boolean(login);
};

export const getToDosSession = async () => {
  await sleep(1000);
  const todos = await localStorage.getItem("todos");
  return todos;
};

export const setToDosSession = async (todos) => {
  await sleep(1000);
  await localStorage.setItem("todos", JSON.stringify(todos));
};
