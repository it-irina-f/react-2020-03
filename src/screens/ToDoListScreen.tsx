import React from "react";
import { ToDoList } from "@/components";
import { RouteComponentProps } from "react-router-dom";
import { logout } from "@/api/auth";
import { authorizedOnlyHoc } from "@/utils/authorizedOnlyHOC";

interface RouteParams {
  name: string;
}

class RawUserScreen extends React.PureComponent<
  RouteComponentProps<RouteParams>,
  {}
> {
  logout = async () => {
    await logout();
    this.props.history.push("/auth");
  };
  render() {
    return (
      <>
        <div>
          <h1>Добро пожаловать, {this.props.match.params.name}!</h1>
          <button onClick={this.logout}>Выход</button>
        </div>
        <ToDoList />
      </>
    );
  }
}

export const ToDoListScreen = authorizedOnlyHoc(RawUserScreen, "/auth");
