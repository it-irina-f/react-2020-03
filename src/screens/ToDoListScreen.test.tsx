import React from "react";
import { ToDoListScreen } from "./ToDoListScreen";
import { shallow } from "enzyme";
import { logout } from "@/api/auth";
import renderer from "react-test-renderer";

const mockHistory = { push: jest.fn() };
jest.mock("react-router-dom", () => ({
  useHistory: () => mockHistory,
}));

jest.mock("@/api/auth", () => ({
  logout: jest.fn(),
}));

jest.useFakeTimers();

describe("ToDoListScreen Test", () => {
  it("renders page ToDoList", () => {
    expect(renderer.create(<ToDoListScreen />).toJSON()).toMatchSnapshot();
  });

  it("click for button LogOut", async () => {
    const screen = shallow(<ToDoListScreen />);

    setTimeout(() => {
      screen.find("Button").simulate("submit");

      expect(logout).toHaveBeenCalledWith();
      expect(mockHistory.push).toHaveBeenCalledWith(`/auth`);
    }, 1200);
  });
});
