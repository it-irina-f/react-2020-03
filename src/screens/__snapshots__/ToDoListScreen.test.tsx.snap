// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoListScreen Test renders page ToDoList 1`] = `
<div>
  loader...
</div>
`;
